ArcadiaTest/DataLayer/TaskRepository.cs

```
@@ -15,109 +15,98 @@ public TaskRepository(ArcadiaTestEntities dbCtx)
+            return this._dbCtx.Tasks.Where(t => t.UserId == userId).ToList().ToDtos();
         }
 
-        public IEnumerable<Task> FindTasksByUserIdAndStatus(int userId, string status)
+        public IEnumerable<TaskDTO> FindTasksByUserIdAndStatus(int userId, string status)
         {
-            return this._dbCtx.Tasks.Where(t => t.UserId == userId && t.Status == status).ToList();
+            return this._dbCtx.Tasks.Where(t => t.UserId == userId && t.Status == status).ToList().ToDtos();
         }
 
-        public Task Save(Task task)
+        public TaskDTO Save(TaskDTO task)
         {
-            var inserted = this._dbCtx.Tasks.Add(task);
+            var taskEntity = new Task();
+            taskEntity.MergeWithDto(task);
+            var inserted = this._dbCtx.Tasks.Add(taskEntity);
             this._dbCtx.SaveChanges();
-            return inserted;
+            return inserted.ToDto();
         }
 
-        public Task UpdateDescription(int taskId, string description)
+        public TaskDTO Update(TaskDTO task)
         {
-            var found = this._dbCtx.Tasks.Where(t => t.Id == taskId).FirstOrDefault();
-            if (found == null)
+            var taskEntity = this.findTaskById(task?.Id ?? 0);
+            if (taskEntity == null)
             {
                 return null;
             }
-            found.Description = description;
+            taskEntity.MergeWithDto(task);
             this._dbCtx.SaveChanges();
-            return found;
+            return taskEntity.ToDto();
         }
 
-        public Task UpdateDueDate(int taskId, DateTime? dueDate)
+        public void Delete(TaskDTO task)
         {
-            var found = this._dbCtx.Tasks.Where(t => t.Id == taskId).FirstOrDefault();
-            if (found == null)
-            {
-                return null;
-            }
-            found.DueDate = dueDate;
+            var taskEntity =this.findTaskById(task?.Id ?? 0);
```

Why do you need to complicated condition? If task is null or task.Id is not set - this method should throw a custom error.